File: implement_queue_with_stacks.py
        
class MyQueue:

    def __init__(self):
        self.st1 = []
        self.st2 = []
        

    def push(self, x: int) -> None:
        self.st1.append(x)

    def pop(self) -> int:
        while len(self.st1) != 0 :
            self.st2.append(self.st1.pop())
        ans =  self.st2.pop()    
        while len(self.st2) !=0:
            self.st1.append(self.st2.pop())
        return ans
        
    def peek(self) -> int:
        return self.st1[0]

    def empty(self) -> bool:
        if len(self.st1) > 0:
            return False
        return True


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: Implement_queue_with_stacks.java
class MyQueue {
    Stack<Integer> stack1 = new Stack<>();
    Stack<Integer> stack2 = new Stack<>();

    public MyQueue() {

    }

    public void push(int x) {
        stack1.push(x);

    }

    public int pop() {
        if(stack2.empty()){
            while(!stack1.empty()){
                int stack1Pop = stack1.pop();
                stack2.push(stack1Pop);
            }

        }
        return stack2.pop();
    }

    public int peek() {
        //     if(!stack2.empty()){
        //         stack2.peek();
        //     }else {
        // while(!stack1.empty()){
        //     int stack1Pop = stack1.pop();
        //     stack2.push(stack1Pop);
        // }
        //         return stack2.peek();
        if(stack2.empty()){
            while(!stack1.empty()){
                int stack1Pop = stack1.pop();
                stack2.push(stack1Pop);
            }
        }
        return stack2.peek();



    }

    public boolean empty() {
        return (stack2.empty() && stack1.empty());

    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */